### PR TITLE
BXMSDOC-2294 PAM Clustering on EAP

### PR DESCRIPTION
**Pull request by emmurphy1**
This pull request includes the following jiras:
[BXMSDCO-2465: PAM Cluster: Clustering the runtime environment](https://issues.jboss.org/browse/BXMSDOC-2465)
[BXMSDCO-2466: PAM Cluster: Clustering tfor the environment](https://issues.jboss.org/browse/BXMSDOC-2466)

It replaces [PR 55](https://github.com/emmurphy1/kie-docs/pull/55).

### CODE REVIEW

File: docs/product-installation-guide/src/main/asciidoc/clustering-runtime-managed-proc.adoc [hunk 1]
```diff
@@ -8,7 +8,7 @@
 . Set up a {EAP} 7.1 cluster as described in  https://access.redhat.com/documentation/en-us/reference_architectures/2017/html-single/configuring_a_red_hat_jboss_eap_7_cluster/[_Configuring a {EAP} 7 Cluster_].
```
Comment (martinweiler) [inline docs/product-installation-guide/src/main/asciidoc/clustering-runtime-managed-proc.adoc]: This is a huge document. AFAIK, clustering the runtime environment only requires:
* shared DB settings
* clustered EJB timer implementation
* kie container alignment

So asking customers to follow this long document just to meet these two requirements is quite an overhead IMO. Moreover, I'm not even sure if this results in clustered EJB timers (it is not explicitly stated). @maciejs has the final word, but I'd think that the whole clustering section could be replaced.

File: docs/product-installation-guide/src/main/asciidoc/clustering-runtime-managed-proc.adoc [hunk 2]
```diff
@@ -8,7 +8,7 @@
...
 . Install and configure Elasticsearch as described in <<clustering-elasticsearch-proc_{context}>>.
 . Install and configure AMQ Broker as described in <<clustering-artemis-activate-proc_{context}>>.
```
Comment (martinweiler) [inline docs/product-installation-guide/src/main/asciidoc/clustering-runtime-managed-proc.adoc]: Not required for runtime clustering IMO

File: docs/product-installation-guide/src/main/asciidoc/clustering-runtime-managed-proc.adoc (hunk 1 above)
Comment (maciejs) [reply docs/product-installation-guide/src/main/asciidoc/clustering-runtime-managed-proc.adoc]: No idea on why you mentioned me here.

Comment (mswiderski) [reply docs/product-installation-guide/src/main/asciidoc/clustering-runtime-managed-proc.adoc]: @maciejs I think @martinweiler meant to mention me instead, anyway thanks for replying

@martinweiler there might be many aspects of the EAP that is good to be taken into consideration for cluster setup and thus it makes sense to refer to official cluster guide. One of such things is the JMS provider that makes sense to have clustered.

**Review by porcelli — APPROVED** [review]

File: docs/product-installation-guide/src/main/asciidoc/clustering-dev-proc.adoc [hunk 3]
```diff
@@ -1,17 +1,17 @@
...
+Developers use {PRODUCT} to create rules and processes that users then run to assist with decision making. The primary benefit of clustering {PRODUCT} development environments is high availability and enhanced collaboration. For example, if clustering is configured and Developer X is authoring rules on {CENTRAL} Node1 then Developer Y, who is working on {CENTRAL} Node2, can see the rules authored by Developer X as they are created. Furthermore, all of the developers working on individual nodes of the cluster will see the same synchronized content. In addition, clustering provides high availability in the rules development environment. If Developer X is working on Node1 and that node fails, Developer X's work is preserved and visible on any other node of the cluster.
```
Comment (sterobin) [inline docs/product-installation-guide/src/main/asciidoc/clustering-dev-proc.adoc]: This is a good time to get your thoughts on a recurring terminology issue. You don't have to change anything if you don't want, but wanted to loop you in.

A recurring terminology issue is what to call the "stuff" that users create in Central. The blanket term to date has been "business assets" and "assets", which encompass rules, processes, tasks, etc. Specifying "rules and processes" is tricky because it's a reuse issue (doesn't apply to DM, which has only rules). What I've found helpful is to subtly define assets in passing the first time, then use assets when referring collectively after that. Christy and I have been in discussion about this a lot and something I've been meaning to ask you about. 

So here, it could be "to create business assets, such as rules, processes, and tasks, that users then run..." This first mention could be ifdefed, then subsquent mentions could just be to "assets" or "business assets".

Now, this poses a problem: "assets" is vague. Assets are what these things are called in Central, yes, but everything else is also called assets (processes, tasks, decision tables, drl files, all alike). So when you execute an "asset", you're really executing rules, processes, or tasks (I think that's it, maybe more). To make things more complicated, what is a blanket term for complete "decision services + process applications + task lists"? These are the finished products of the "assets" and are what we are also executed as a whole.

In short, these are the terms I had to come up with in the Packaging doc with no alternative and that we should decide on as a team soon (post GA):
- rules, processes, tasks, decision tables, data objects, bla bla = "business assets" / "assets"
- decision services, process applications, task lists = "services"

The "services" remedy is not great and SMEs only let me use that if I define it upon first mention, and even then they're not totally sold on it but don't have suggestions. I don't know what else to do and would love your input before asking the team. Maybe a call sometime.

File: docs/product-installation-guide/src/main/asciidoc/clustering-dev-proc.adoc [hunk 4]
```diff
@@ -1,17 +1,17 @@
...
 
 [NOTE]
 ====
-You can also use this configuration for runtime environemtns if you want to use {CENTRAL} to manage the {KIE_SERVER}s in the cluster.
+You can also use this configuration for runtime environments if you want to use {CENTRAL} to manage the {KIE_SERVER}s in the cluster.
 ====
 
 .Prerequiste
-* An NFS server with a mounted partition accessible to an {EAP} user is available.
+* An NFS server with a mounted partition accessible to a {EAP} user is available.
```
Comment (sterobin) [inline docs/product-installation-guide/src/main/asciidoc/clustering-dev-proc.adoc]: No bullet for single bulletted item. Comes up elsewhere as well. Check around.

File: docs/product-installation-guide/src/main/asciidoc/clustering-elasticsearch-proc.adoc [hunk 5]
```diff
@@ -1,34 +1,27 @@
...
+Elasticsearch is a highly scalable open source full-text search and analytics engine. It enables you to store, search, and analyze high volumes of data quickly and in near real time.  In a {PRODUCT} clustered environment, it enables you to perform complex and efficient searches across cluster nodes. 
 
-.Prerequisite
-A {EAP} 7.1 cluster is configured as described in  https://access.redhat.com/documentation/en-us/reference_architectures/2017/html-single/configuring_a_red_hat_jboss_eap_7_cluster/[_Configuring a {EAP} 7 Cluster_].
+Set up Elasticsearch on one node of the cluster.
+
+[NOTE]
+====
+This procedure describes how to set up Elasticseach for a simplified, non-high availability environement. For high availability production environments, see the https://www.elastic.co/guide/en/elasticsearch/guide/current/deploy.html[Production Deployment] section of the Elasticsearch documentation.
```
Comment (sterobin) [inline docs/product-installation-guide/src/main/asciidoc/clustering-elasticsearch-proc.adoc]: Super trivial (comes up a couple of times): "non-high availability" should actually use an en dash instead of a high (hyphens with compound words in Chicago), but I haven't seen how to easily do it yet in Atom (doesn't come up enough). Em dashes are fine in Atom, but no en dashes that I see yet.

File: docs/product-installation-guide/src/main/asciidoc/clustering-elasticsearch-proc.adoc [hunk 6]
```diff
@@ -1,34 +1,27 @@
...
-+
-`_ELASTICSEARCH_HOME_/bin/elasticsearch-plugin install x-pack`
-+
-* On Windows, enter:
-+
-`_ELASTICSEARCH_HOME_\bin\elasticsearch-plugin.bat install x-pack`
-
 . Run Elasticsearch:
 +
 * On Linux or UNIX-based systems, enter:
```
Comment (sterobin) [inline docs/product-installation-guide/src/main/asciidoc/clustering-elasticsearch-proc.adoc:22]: Make the commands that follow code blocks, for consistency.

File: docs/product-installation-guide/src/main/asciidoc/clustering-dm-install-proc.adoc [hunk 7]
```diff
@@ -8,10 +8,10 @@ A {EAP} 7.1 cluster is configured and Elasticsearch and AMQ Broker 7.1 are insta
 
 .Procedure
 
-. Install {CENTRAL} on each node of the cluster. See _INSTALLING AND CONFIGURING RED HAT PROCESS AUTOMATION MANAGER ON EAP 7.1_ for information about installing {CENTRAL}.
-. On each node of the cluster, edit the following properties in the `<EAP_HOME>/standalone/configuration/standalone.xml` file, where:
-* `<NIOGIT_DIR>` is the location git repositories stored under ``.niogit`, on an NFS mounted partition.
-* `<MAIN_NODE_IP>` is the IP address of the node where Elasticseach and AMQ Broker are installed.
+. Install {CENTRAL} on each node of the cluster. See _Installing and configuring {PRODUCT} on {EAP} 7.1_ for information about installing {CENTRAL}.
+. On each node of the cluster, edit the following properties in the `<EAP_HOME>/domain/configuration/domain.xml` file, where:
+* `<NIOGIT_DIR>` is the location of Git repositories stored under ``.niogit`, on an NFS mounted partition.
```
Comment (sterobin) [inline docs/product-installation-guide/src/main/asciidoc/clustering-dm-install-proc.adoc]: Stray single quote in `.niogit`.

File: docs/product-installation-guide/src/main/asciidoc/clustering-dm-install-proc.adoc [hunk 8]
```diff
@@ -8,10 +8,10 @@ A {EAP} 7.1 cluster is configured and Elasticsearch and AMQ Broker 7.1 are insta
 
 .Procedure
 
-. Install {CENTRAL} on each node of the cluster. See _INSTALLING AND CONFIGURING RED HAT PROCESS AUTOMATION MANAGER ON EAP 7.1_ for information about installing {CENTRAL}.
-. On each node of the cluster, edit the following properties in the `<EAP_HOME>/standalone/configuration/standalone.xml` file, where:
-* `<NIOGIT_DIR>` is the location git repositories stored under ``.niogit`, on an NFS mounted partition.
-* `<MAIN_NODE_IP>` is the IP address of the node where Elasticseach and AMQ Broker are installed.
+. Install {CENTRAL} on each node of the cluster. See _Installing and configuring {PRODUCT} on {EAP} 7.1_ for information about installing {CENTRAL}.
+. On each node of the cluster, edit the following properties in the `<EAP_HOME>/domain/configuration/domain.xml` file, where:
```
Comment (sterobin) [inline docs/product-installation-guide/src/main/asciidoc/clustering-dm-install-proc.adoc]: Comes up multiple times. A case where I personally am great with this and see this as pretty standard and clear. But I think IBM is against this saying "where:" with a list of variables, with the same logic/rule of not ending incomplete sentences with a colon. I think they would probably just end with ".. file:" with the list of things to edit. In this case the "where" is possibly just unnecessary. In other cases, I think IBM would say "Replace the following values:" or the like. Again, using "where:" followed by replacable variables is clear and standard in the industry, and is even recommended in other tech style manuals (Sun Microsystems and Microsoft for instance both do). I'm just making the IBM plug, much to your and my dismay :)

File: docs/product-installation-guide/src/main/asciidoc/clustering-dm-install-proc.adoc [hunk 9]
```diff
@@ -20,24 +20,24 @@ A {EAP} 7.1 cluster is configured and Elasticsearch and AMQ Broker 7.1 are insta
...
-. On the node where Elasticseach and AMQ Broker are installed, in {EAP} create a user with the `admin` role and a username and password that matches the values of `<AMQ_BROKER_USER>` and `<AMQ_BROKER_PASSWORD>`:
+. On the nodes where Elasticseach and AMQ Broker are installed, in {EAP} create a user with the `admin` role and a username and password that matches the values of `<AMQ_BROKER_USER>` and `<AMQ_BROKER_PASSWORD>`:
 +
 [source,bash]
 ----
-$ <MAIN_NODE_EAP_HOME>/bin/./add-user.sh -a --user <AMQ_BROKER_USER> --password <AMQ_BROKER_USER> --role admin
+$ <EAP_HOME>/bin/add-user.sh -a --user <AMQ_BROKER_USER> --password <AMQ_BROKER_USER> --role admin
 ----
 
 .  To start {CENTRAL}, enter one of the following commands on each node of the cluster:
```
Comment (sterobin) [inline docs/product-installation-guide/src/main/asciidoc/clustering-dm-install-proc.adoc]: I'd say "Enter one of the following commands on each node of the cluster to start Business Central:" This is consistent with another section on AMQ, and is more recommended practice in steps (avoid "To do something" as a step).

File: docs/product-installation-guide/src/main/asciidoc/clustering-ks-runtime-standalone-proc.adoc [hunk 10]
```diff
@@ -1,11 +1,39 @@
...
-* An NFS server with a mounted partition accessible to an {EAP} user is available.
+* An NFS server with a mounted partition accessible to a {EAP} user is available.
 
 .Procedure
 . Set up a {EAP} 7.1 cluster as described in  https://access.redhat.com/documentation/en-us/reference_architectures/2017/html-single/configuring_a_red_hat_jboss_eap_7_cluster/[_Configuring a {EAP} 7 Cluster_].
-. Install and configure Elasticsearch as described in <<clustering-elasticsearch-proc_{context}>>.
-. Install and configure AMQ Broker as described in <<clustering-artemis-activate-proc_{context}>>.
-. Install and configure {CENTRAL} as described in <<clustering-kie-server-install-proc_{context}>>.
+. Install {KIE_SERVER} on each node of the cluster as decribed in _Installing and configuring {PRODUCT} on {EAP} 7.1_.
+. If your environment is connected to Smart Router, on the java process running Smart Router do the following:
```
Comment (sterobin) [inline docs/product-installation-guide/src/main/asciidoc/clustering-ks-runtime-standalone-proc.adoc]: Cap "Java". To avoid the "following:" issue in IBM (if you care), I'd reword "do the following on the Java process running Smart Router:" Both suggestions here apply to the next step also.

**Review by sterobin — COMMENTED** [review]
@emmurphy1, peer review done. Great as usual. Trivial suggestions.

File: docs/product-installation-guide/src/main/asciidoc/clustering-custom-config-proc.adoc [hunk 11]
```diff
@@ -0,0 +1,67 @@
...
+----
+./domain.bat
+----
+
+. Check that the nodes are available.
+
+Deploy the {CENTRAL} application to your servers:
+
+ifdef::PAM[]
+. Change the predefined persistence of the application to the required database (PostgreSQL): in `persistence.xml` change the following:
```
Comment (mswiderski) [inline docs/product-installation-guide/src/main/asciidoc/clustering-custom-config-proc.adoc:47]: there is no need to modify persistence.xml file any more in v7 - actually it is embedded in a jar file so it won't be easy to actually modify it. Instead following system properties should be set to direct to data source and hibernate dialect:

```
org.kie.server.persistence.ds
org.kie.server.persistence.dialect
```

**Review by mswiderski — APPROVED** [review]
overall looks very good to me, just one comment about editing persistence.xml which is no longer valid for v7
